pkg/ingress/controller/openstack/neutron.go

```diff
@@ -100,9 +100,41 @@ func (os *OpenStack) EnsureFloatingIP(needDelete bool, portID string, floatingIP
 			FloatingNetworkID: floatingIPNetwork,
 			Description:       description,
 		}
-		fip, err = floatingips.Create(os.neutron, floatIPOpts).Extract()
-		if err != nil {
-			return "", err
+		if existingfloatingIP != "" {
+			// try to find fip
+			opts := &floatingips.ListOpts{
+				FloatingIP:        existingfloatingIP,
+				FloatingNetworkID: floatingIPNetwork,
+			}
+			allPages, err := floatingips.List(os.neutron, opts).AllPages()
+			if err != nil {
+				return "", err
+			}
+			osFips, err := floatingips.ExtractFloatingIPs(allPages)
+			if err != nil {
+				return "", err
+			}
+			if len(osFips) != 1 {
+				return "", err
+			}
+			// check if fip is used
+			if osFips[0].PortID != "" {
+				return "", fmt.Errorf("floating IP %s already used by port %s", osFips[0].FloatingIP, osFips[0].PortID)
+			}
```

Comment:
What if `osFips[0].PortID == portID` already? That should be considered success, right?